drivers/ethernet/oa_tc6.c

```diff
@@ -235,59 +265,413 @@ int oa_tc6_set_protected_ctrl(struct oa_tc6 *tc6, bool prote)
 	return 0;
 }
 
-int oa_tc6_send_chunks(struct oa_tc6 *tc6, struct net_pkt *pkt)
+static int oa_tc6_submit_rx_pkt(struct oa_tc6 *tc6)
 {
-	uint16_t len = net_pkt_get_len(pkt);
-	uint8_t oa_tx[tc6->cps];
-	uint32_t hdr, ftr;
-	uint8_t chunks, i;
 	int ret;
 
-	if (len == 0) {
-		return -ENODATA;
+	ret = net_recv_data(tc6->iface, tc6->rx_pkt);
+	if (ret < 0) {
+		eth_stats_update_errors_rx(tc6->iface);
+		net_pkt_unref(tc6->rx_pkt);
+		return ret;
 	}
+	eth_stats_update_pkts_rx(tc6->iface);
+	tc6->rx_pkt = NULL;
 
-	chunks = len / tc6->cps;
-	if (len % tc6->cps) {
-		chunks++;
+	return 0;
+}
+
+static int oa_tc6_update_rx_pkt(struct oa_tc6 *tc6, uint8_t *payload, uint8_t length)
+{
+	int ret;
+
+	ret = net_pkt_write(tc6->rx_pkt, payload, length);
+	if (ret < 0) {
+		net_pkt_unref(tc6->rx_pkt);
+		eth_stats_update_errors_rx(tc6->iface);
+		return ret;
 	}
+	eth_stats_update_bytes_rx(tc6->iface, length);
 
-	/* Check if LAN865x has any free internal buffer space */
-	if (chunks > tc6->txc) {
-		return -EIO;
+	return 0;
+}
+
+static int oa_tc6_allocate_rx_pkt(struct oa_tc6 *tc6)
+{
+	tc6->rx_pkt = net_pkt_rx_alloc_with_buffer(tc6->iface, OA_TC6_ETH_BUFFER_SIZE, AF_UNSPEC, 0,
```

Comment:
I suspect that not all uC will be able to perform successfully this operation (more detailed comment follow)